src/wwwroot/startupscripts/startagent-win.cmd

```diff
@@ -7,6 +7,23 @@ taskkill -f -im dotnet.exe
 taskkill -f -im vbcscompiler.exe
 taskkill -f -im devenv.exe
 rmdir /S /Q %WORKSPACEPATH%
+
+REM Sometimes killing the processes above fails; there could also be surprise new processes.
+REM As the disk here is big and the consequences of a dirty workspace are bad, let's make a new one.
+REM We'll always try to delete the existing ones in ascending order so if it's possible, we'll eventually clean up.
+if exist "%WORKSPACEPATH%" (
+set /a suffix=0
+:while
+set candidateworkspacepath=%WORKSPACEPATH%.%suffix%
+echo Trying %candidateworkspacepath%... 
+if exist "%candidateworkspacepath%" (
```

Comment:
Will this work past index 0? I think you may need to enable delayed variable expansion and use !! variables in order for this to work, but i"m not positive.

Reply:
I tested this up to 50 folders, this version definitely does what you'd expect...